Makefile

```diff
@@ -193,6 +193,15 @@ errdoc: install-go-tools
 	@echo "generator errors.toml"
 	./scripts/check-errdoc.sh
 
+docker-build-test:
+	$(eval DOCKER_PS_EXIT_CODE=$(shell docker ps > /dev/null 2>&1 ; echo $$?))
+	@if [ $(DOCKER_PS_EXIT_CODE) -eq 0 ]; then \
+	docker build --no-cache -t tikv/pd .; \
+	else \
+	echo "Encountered problem while invoking docker cli. Is docker up and running? Failing docker-build-test. Check exit code for details."; \
```

Comment:
```suggestion
	echo "Encountered problem while invoking docker cli. Is the docker daemon running?"; \
```

Reply:
Hey, @Yisaer, sometimes when the users don't have permissions to run docker, they run docker commands using `sudo docker`. "Check exit code for details." will make them check exit code and hence point them to the correct issue in this case.

Also, "Is docker up and running?" feels a better error msg to me than "Is the docker daemon running?" because we are specifically using docker's cli client, and even if the docker daemon is up and running, there might be issues with the cli client which could fail the test. "Is docker up and running?" covers every scenario where there could be a fault from docker side of things.

For "Failing docker-build-test." I have no opinion. I can remove it if you want.

Please review my comments above. I will make the changes after you give the feedback again.

Reply:
`Is the docker daemon running` is the default response for docker cli when docker daemon is not started or running, I just tend to keep the same behavior with docker cli.

Reply:
Do we not want to check for conditions where users don't have permission to run docker?
```
user@server:~$ docker ps
Got permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Get http://%2Fvar%2Frun%2Fdocker.sock/v1.40/containers/json: dial unix /var/run/docker.sock: connect: permission denied
```

Reply:
We can throw a simple response and expose the error message to let the users find the reason themselves.

Reply:
So would this work? Basically we are exposing the error msg but running the `docker build` command in the end anyway.
```makefile
docker-build-test:
	$(eval DOCKER_PS_EXIT_CODE=$(shell docker ps > /dev/null 2>&1 ; echo $$?))
	@if [ $(DOCKER_PS_EXIT_CODE) -ne 0 ]; then \
	echo "Encountered problem while invoking docker cli. Is the docker daemon running?"; \
	fi
	docker build --no-cache -t tikv/pd .
```

When docker daemon is not running, it would give output like this
```
$ make docker-build-test                                                                          
Encountered problem while invoking docker cli. Is the docker daemon running?
docker build --no-cache -t tikv/pd .
Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
make: *** [docker-build-test] Error 1
```

In case docker is not installed, we would get this error:
```
$ make docker-build-test                                                                          
Encountered problem while invoking docker cli. Is the docker daemon running?
docker build --no-cache -t tikv/pd .
/bin/bash: docker: command not found
make: *** [docker-build-test] Error 127
```

Reply:
I think This is ok to me.

Reply:
@Yisaer check now